src/GeneralStateTestsFiller/stPreCompiledContracts/modexpTestsFiller.yml

```diff
@@ -0,0 +1,421 @@
+# These are tests for precomp 5, modexp 
+# (specified in https://github.com/ethereum/EIPs/blob/master/EIPS/eip-198.md)
+#
+# For every case there are two parameters to test:
+# 1. The result
+# 2. The gas cost
+
+modexpTests:
+
+  env:
+    currentCoinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    currentDifficulty: '0x20000'
+    currentGasLimit: "100000000"
+    currentNumber: "1"
+    currentTimestamp: "1000"
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+
+  _info:
+    comment: Ori Pomerantz qbzzt1@gmail.com
+
+
+  pre:
+
+    # Test for cases where all three parameters are <2^256. 
+    cccccccccccccccccccccccccccccccccccccccc:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: |
+       :yul {
+          // Our input values, 20 bytes each
+          // This is not the most efficient use of gas, but 
+          // this is a test. Readability is more important
+          let base := calldataload(0x04)
+          let expV := calldataload(0x24)
+          let modV := calldataload(0x44)
+
+          // Prepare the calldata
+          mstore(0x00, 0x20)
+          mstore(0x20, 0x20)
+          mstore(0x40, 0x20)
+          mstore(0x60, base)
+          mstore(0x80, expV)
+          mstore(0xA0, modV)
+
+          let gas0 := gas()
+          pop(call(gas(), 0x05, 0, 0, 0xC0, 0x100, 0x20))
+          let gas1 := gas()
+          sstore(0, mload(0x100))
+          sstore(1, sub(sub(gas0, gas1), 0x14c))
+       }
+      nonce: 1
+      storage: {}
+
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: 0x
+      nonce: 1
+      storage: {}
+
+
+  transaction:
+    data: 
+    - :label m_n_01 :abi f(uint,uint,uint) 0 0 0
+    - :label m_n_01 :abi f(uint,uint,uint) 0 0 1
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 2
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 4
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 8
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x10
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x20
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x40
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 100
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x80
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x1001
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x100002
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x10000004
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0x1000000008
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0xFFFFFFFFFFFF
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0xFFFFFFFFFFFFFF
+    - :label 0_0_n :abi f(uint,uint,uint) 0 0 0xFFFFFFFFFFFFFFFF
```

Comment:
please double check that abi does encode here properly and the contract gets the value here properly.  u256.

Reply:
Double checked for `-d 0` through `-d 23`. All are fine.